### PR TITLE
[Meerkat] Add sign in activity

### PR DESCRIPTION
**Pull request by younghajung**
This patch adds Google sign in activity for Meerkat server.

Signed-off-by: yh106.jung <yh106.jung@samsung.com>

### CODE REVIEW

**Review by Chikdolman — COMMENTED** [review]
Looks neat. Will you make other PR for sharing tokens? or will it be on this PR?

**Comment by younghajung** [comment]
> Looks neat. Will you make other PR for sharing tokens? or will it be on this PR?

I'll make another PR for authentication.

File: third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatSignInActivity.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        setContentView(R.layout.signin_activity);
+
+        mDetailTextView = findViewById(R.id.detail);
+
+        findViewById(R.id.sign_in_button).setOnClickListener(this);
+        findViewById(R.id.sign_out_button).setOnClickListener(this);
+        findViewById(R.id.start_button).setOnClickListener(this);
+
+        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken("401503586848-3ajf0semvlclbffipcuh7oc7qr2kattk.apps.googleusercontent.com")
```
Comment (djmixkim) [inline third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatSignInActivity.java:57]: Does this token always use same value?
Don't we have any security issue when use hard-coded value here?

File: third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatServerService.java [hunk 2]
```diff
@@ -37,15 +37,15 @@
...
+    private static final String TAG = "MeerkatServerService";
     private static final int MEERKAT_NOTIFICATION_ID = 100;
     private static final String MEERKAT_CHANNEL_ID = "meekat_server";
     private static final String MEERKAT_CHANNEL_GROUP_ID = "meekat";
     private static final String CASTANETS_PACKAGE_NAME = "org.chromium.chrome";
     private static final String ACTION_NOTIFICATION_CLICKED = "com.samsung.android.meerkat.NOTIFICATION_CLICKED";
 
     private static Context sApplicationContext;
-    private static Thread mMainThread;
+    private Thread mMainThread;
```
Comment (djmixkim) [inline third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatServerService.java:48]: Why did you remove "static" here?

File: third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatSignInActivity.java (hunk 1 above)
Comment (younghajung) [reply third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatSignInActivity.java:57]: That is not ID token, that is client ID of Castanet service.

File: third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatServerService.java (hunk 2 above)
Comment (younghajung) [reply third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatServerService.java:48]: `mMainThread` is a member of each instance of MeerkatServerService, so I remove `static` keyword.

**Review by djmixkim — APPROVED** [review]
lgtm
